src/Test/L0/Worker/TaskManagerL0.cs

```diff
@@ -131,5 +132,153 @@ private Stream GetZipStream()
                 }
             }
         }
+
+        //Test the cancellation flow: interrupt download task via HostContext cancellation token.
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public async void TestCancellationOfEnsureTasksExist()
+        {
+            //Arrange
+            using (var hc = new TestHostContext(this))
+            {
+                hc.SetSingleton<IJobServer>(_jobServer.Object);
+                hc.SetSingleton<IConfigurationStore>(_configurationStore.Object);
+                var taskManager = new TaskManager();
+                taskManager.Initialize(hc);
+
+                _configurationStore.Setup(x => x.GetSettings()).Returns(
+                 new AgentSettings
+                 {
+                     WorkFolder = WorkFolderName
```

Comment:
FYI - Also a lot of this stuff is duplicate for each test. I think I created a common function in my branch. Don't worry about it here, I'll fix it after I rebase. Just FYI for future.